Tribler/Policies/BoostingSource.py

```diff
@@ -225,22 +223,23 @@ def showtorrent(torrent):
             """
             assembly torrent data, call the callback
             """
-            if torrent.files:
+            infohash = torrent.infohash
+            if torrent.get_files() and infohash in self.unavail_torrent:
                 if len(self.torrents) >= self.max_torrents:
                     self._logger.debug("Max torrents in source reached. Not adding %s", torrent.infohash)
                     del self.unavail_torrent[torrent.infohash]
                     return
 
-                infohash = torrent.infohash
                 self._logger.debug("[ChannelSource] Got torrent %s", hexlify(infohash))
                 self.torrents[infohash] = {}
-                self.torrents[infohash]['name'] = torrent.name
-                self.torrents[infohash]['metainfo'] = torrent.tdef
-                self.torrents[infohash]['creation_date'] = torrent.creation_date
-                self.torrents[infohash]['length'] = torrent.tdef.get_length()
-                self.torrents[infohash]['num_files'] = len(torrent.files)
-                self.torrents[infohash]['num_seeders'] = torrent.swarminfo[0] or 0
-                self.torrents[infohash]['num_leechers'] = torrent.swarminfo[1] or 0
+                self.torrents[infohash]['name'] = torrent.get_name()
+                self.torrents[infohash]['metainfo'] = torrent
+                self.torrents[infohash]['creation_date'] = torrent.get_creation_date()
+                self.torrents[infohash]['length'] = torrent.get_length()
+                self.torrents[infohash]['num_files'] = len(torrent.get_files())
+                #TODO(ardhi) get seeder/leecher from db
+                self.torrents[infohash]['num_seeders'] = 0
```

Comment:
It a default/start value. It will be replaced later when scraping the torrent. It will be better if I can provide number of seeder/leecher right away from database, but I can't find any way to do it now.